packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -784,8 +785,9 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
                 callSite,
                 pkg: packagePathToTelemetryProperty(this.pkg),
             }, error);
-
-            if (this.throwOnTombstoneUsage) {
+            // Always log an error when tombstoned data store is used. However, throw an error only if
+            // throwOnTombstoneUsage is set and the client is not a summarizer.
+            if (this.throwOnTombstoneUsage && this.clientDetails.type !== summarizerClientType) {
```

Comment:
You should use `this.clientDetails.type !== summarizerClientType` while setting `throwOnTombstoneUsage` in the constructor. That way this check is not required everytime and a single boolean check here will be enough.
Same commet for datastores.ts